src/client/libcurve_file.cpp

```diff
@@ -517,6 +531,37 @@ int FileClient::Recover(const std::string &filename, const UserInfo_t &userinfo,
     return -ret;
 }
 
+int FileClient::StatFile(int fd, FileStatInfo *finfo) {
+    {
+        ReadLockGuard lk(rwlock_);
+        auto iter = fileserviceMap_.find(fd);
+        if (iter == fileserviceMap_.end()) {
+            LOG(ERROR) << "StatFile failed not found fd = " << fd;
+            return -LIBCURVE_ERROR::FAILED;
+        }
+    }
+
+    FileInstance *instance = fileserviceMap_[fd];
+
+    FInfo_t fi = instance->GetCurrentFileInfo();
```

Comment:
these two lines of code should also be protected by the lock.

Reply:
done